### PR TITLE
[AIX][TLS][clang]  Add -maix-small-local-dynamic-tls clang option

### PR DESCRIPTION
**Pull request by orcguru**
This patch adds the clang portion of an AIX-specific option to inform the
compiler that it can use a faster access sequence for the local-dynamic
TLS model (formally named aix-small-local-dynamic-tls).

This patch mainly references Amy's work on small local-exec TLS support.

### CODE REVIEW

File: clang/test/Driver/aix-small-local-exec-tls.c [hunk 1]
```diff
@@ -6,6 +6,9 @@
 // RUN: %clang -target powerpc64-unknown-aix -maix-small-local-exec-tls -S -emit-llvm \
 // RUN:    %s -o - | FileCheck %s --check-prefix=CHECK-AIX_SMALL_LOCALEXEC_TLS
 
+// RUN: %clang -target powerpc64-unknown-aix -maix-small-local-dynamic-tls -S -emit-llvm \
```
Comment (chenzheng1030) [inline clang/test/Driver/aix-small-local-exec-tls.c:9]: nit: maybe we can rename this file?

**Review by chenzheng1030 — APPROVED** [review]
LGTM with one nit. Thanks for adding this support.

File: clang/test/Driver/aix-small-local-exec-tls.c (hunk 1 above)
Comment (amy-kwan) [reply clang/test/Driver/aix-small-local-exec-tls.c:9]: I agree here.

**Review by amy-kwan — APPROVED** [review]
LGTM as well.

**Pull request merged**